modules/benchmark/iperf3.c

```diff
@@ -0,0 +1,123 @@
+/*
+ *    HardInfo - Displays System Information
+ *    Copyright (C) 2003-2020 Leandro A. F. Pereira <leandro@hardinfo.org>
+ *    Copyright (C) 2020 Burt P. <pburt0@gmail.com>
+ *
+ *    This program is free software; you can redistribute it and/or modify
+ *    it under the terms of the GNU General Public License as published by
+ *    the Free Software Foundation, version 2.
+ *
+ *    This program is distributed in the hope that it will be useful,
+ *    but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *    GNU General Public License for more details.
+ *
+ *    You should have received a copy of the GNU General Public License
+ *    along with this program; if not, write to the Free Software
+ *    Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA  02110-1301 USA
+ */
+
+#include "hardinfo.h"
+#include "benchmark.h"
+#include <json-glib/json-glib.h>
+#include <math.h>
+
+static int iperf3_version() {
+    int ret = -1.0;
+    int v1 = 0, v2 = 0, v3 = 0, mc = 0;
+    gboolean spawned;
+    gchar *out, *err, *p, *next_nl;
+
+    spawned = g_spawn_command_line_sync("iperf3 --version",
+            &out, &err, NULL, NULL);
+    if (spawned) {
+        ret = 0;
+        p = out;
+        while(next_nl = strchr(p, '\n')) {
+            *next_nl = 0;
+            /* version */
+            mc = sscanf(p, "iperf %d.%d", &v1, &v2);
+            if (mc >= 1) {
+                ret += v1 * 1000000;
+                ret += v2 * 1000;
+                ret += v3;
+                break;
+            }
+            p = next_nl + 1;
+        }
+        g_free(out);
+        g_free(err);
+    }
+    return ret;
+}
+
+static gboolean iperf3_server() {
+    const char* argv[] = {
+        "iperf3", "-s", "--port", "61981", "--one-off", NULL };
+    return g_spawn_async(NULL, (char**)argv, NULL,
+        G_SPAWN_STDERR_TO_DEV_NULL | G_SPAWN_STDOUT_TO_DEV_NULL | G_SPAWN_SEARCH_PATH,
+        NULL, NULL, NULL, NULL);
+}
+
+static double _get_double(JsonParser *j, const char* path) {
+    double r = NAN;
+    GError *e = NULL;
+    JsonNode *root = json_parser_get_root(j);
```

Comment:
Building this should be conditional on finding the JSON library -- it's an optional dependency.